### DISCUSSION TITLE
The following static media file failed to load: setmode.js

### DISCUSSION THREAD

**Discussion by mad-fj** [netbox-community/netbox#11002] [Q&A]
### NetBox version

v3.3.8

### Python version

3.10

### Steps to Reproduce

1. Update from v3.3.7 to v3.3.8
2. Open netbox in browser

### Expected Behavior

No static media error

### Observed Behavior

![grafik](https://user-images.githubusercontent.com/20044393/203387092-5fdcced8-14b6-41f7-8474-7fceb61cb5ef.png)

**Comment by candlerb** [comment]
Your "steps to reproduce" doesn't work for me, as I upgraded from 3.3.7 to 3.3.8 without issues.

Did you use "git pull" to upgrade, or are you working with release tarballs?

**Comment by mad-fj** [comment]
Encountered that this problem only occures, when `BASE_PATH` is set.
e.g. `BASE_PATH=/`
Without setting it, everything works well

related to  #10837

**Reply by candlerb** [reply]
As per documentation, BASE_PATH needs to be set to empty string when not used.  ~~`BASE_PATH = '/'` may have worked by accident before.~~

```
BASE_PATH = getattr(configuration, 'BASE_PATH', '')
if BASE_PATH:
    BASE_PATH = BASE_PATH.strip('/') + '/'  # Enforce trailing slash only
```

But you're right: when I set `BASE_PATH = '/'` on my own system, I get a 404 error for

```
/%2Fmedia-failure/?filename=setmode.js
```

This value must be being picked up somewhere *prior* to BASE_PATH being sanitized, but I have no idea where.

Still: for now, the solution is to set `BASE_PATH = ''` instead of `BASE_PATH = '/'`

**Reply by candlerb** [reply]
OK, I understand it now:

```
>>> BASE_PATH=''
>>> if BASE_PATH:
...     BASE_PATH = BASE_PATH.strip('/') + '/'  # Enforce trailing slash only
...
>>> BASE_PATH
''
```

That is: if BASE_PATH is empty string, it remains empty string.  If it's not empty, then it has a trailing slash but no leading slash.

Many URLs are made relative to this: e.g.

```
MEDIA_URL = '/{}media/'.format(BASE_PATH)
```

which would have a leading double slash if `BASE_PATH = '/'` - hence this is an invalid setting.

Netbox could be made more robust against this particular configuration error, e.g. (untested)

```
BASE_PATH = getattr(configuration, 'BASE_PATH', '').strip('/')
if BASE_PATH:
    BASE_PATH = BASE_PATH + '/'  # Enforce trailing slash only
```

**Reply by kirillmuravyev** [reply]
Hello,
I have encountered the same issue with BASE_PATH = ''
What could be the problem?

![image](https://user-images.githubusercontent.com/40544508/232077730-ac21f02d-ddd1-43d0-90cd-b083f2419f9f.png)

**Reply by kirillmuravyev** [reply]
found the problem.
Ngins configuration comes with "user nginx".
Had to change it to "netbox" as netbox user has proper permissions to netbox folders

**Reply by Alemondemon** [reply]
> found the problem. Ngins configuration comes with "user nginx". Had to change it to "netbox" as netbox user has proper permissions to netbox folders

I don't know which configuration file has this issue of the wrong user, does anyone happen to know?

**Reply by kirillmuravyev** [reply]
Look here in the official guide:
https://netbox.poc.mars-ad.net/static/docs/installation/5-http-server/

be sure that netbox user has all the permissions to /opt/netbox/ folder and all subfolders

**Reply by Alemondemon** [reply]
Thank you, I can't seem to find the correct file where the user may need to be set to netbox

**Reply by kirillmuravyev** [reply]
it is nginx.conf file

**Reply by Alemondemon** [reply]
Found it, and it only one place for the user?

**Reply by kirillmuravyev** [reply]
that's right.
Essentially user in nginx.conf file should the one the has all the permissions to /opt/netbox folder and subfolders

**Reply by mbueso** [reply]
Hi am having the same issue:

![image](https://github.com/netbox-community/netbox/assets/88097763/20184208-3809-48e6-be09-e2b4a9abfea6)

netbox is the owner of the following path since only those are mentioned in the documentation:

![image](https://github.com/netbox-community/netbox/assets/88097763/87a1a457-e1a8-4561-ad74-301e895efa39)

also is not clear why this behavior happened since it was working two hours before, and no changes were made to the configuration or to the nginx.conf file, the user on the file is set to nginx, but as mentioned, it was working before, will appreciate any advice on what is needed to troubleshoot this error

**Comment by mbueso** [comment]
Hi am having the same issue:

image

netbox is the owner of the following path since only those are mentioned in the documentation:

image

also is not clear why this behavior happened since it was working two hours before, and no changes were made to the configuration or to the nginx.conf file, the user on the file is set to nginx, but as mentioned, it was working before, will appreciate any advice on what is needed to troubleshoot this error

**Reply by IainACNET** [reply]
Had the same issue on a Centos 9 test box, I got it working by following the install instructions to the end, then adding an include to /etc/nginx/nginx.conf file

under http 

`include /etc/nginx/sites-enabled/netbox;`

Then I commented out both http and https servers in the nginx config file, the import pulls in the server settings from the netbox file, after that nextbox started working as expected for https.

finally for the redirct to work I had to remove :80 at the bottom of the file /etc/nginx/sites-available/netbox 

It's the only way I could get it working, and this is a test box only running netbox so I wasn't concerned about breaking anything else

**Comment by mpcore77** [comment]
Hi
just fyi, ran into same issue today, reason here was just installation path:
instead of /opt/netbox as mentioned in the manual, I used /opt/netbox/current
so nginx config file for netbox needs to be adjusted, too, of course...

from 
`    location /static/ {
        alias /opt/netbox/netbox/static/;
}
`

to (in my case)
`    location /static/ {
        alias /opt/netbox/current/netbox/static/;
}
`
maybe this will help somebody :)
